client/authority-discovery/src/lib.rs

```diff
@@ -78,6 +78,11 @@ pub struct WorkerConfig {
 	///
 	/// Defaults to `true` to avoid the surprise factor.
 	pub publish_non_global_ips: bool,
+
+	/// Reject authority discovery records that are not signed by their network identity (PeerId)
+	///
+	/// Defaults to `false` to provide compatibility with old versions
```

Comment:
IMO this should default to `true` to have new networks enabled this by default.

Reply:
That would cause silent breaking for existing networks, which use `Default::default()` in their `sc-authority-discovery::Worker` instantiation. I would be more conservative here and reduce gotchas in the ecosystem.

Reply:
I think no one uses this, but fine.

Then we should document this in the docs of the crate?